cpu/esp32/include/sdkconfig_esp32s3.h

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2022 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp32
+ * @{
+ *
+ * @file
+ * @brief       SDK configuration used by the ESP-IDF for ESP32-S3 SoC variant (family)
+ *
+ * The SDK configuration can be partially overridden by application-specific
+ * board configuration.
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#ifndef SDKCONFIG_ESP32S3_H
+#define SDKCONFIG_ESP32S3_H
+
+#ifndef DOXYGEN
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name ESP32-S3 specific clock configuration
+ * @{
+ */
+
+/* Mapping of Kconfig defines to the respective enumeration values */
+#if CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_2
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     2
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_5
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     5
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_10
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     10
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_20
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     20
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_40
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     40
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_80
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     80
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_160
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     160
+#elif CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ_240
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     240
+#endif
+
+/**
+ * @brief Defines the CPU frequency [values = 2, 5, 10, 10, 40, 80, 160, 240]
+ */
+#ifndef CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ
+#define CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ     80
+#endif
+/**
+ * @brief Mapping configured ESP32 default clock to CLOCK_CORECLOCK define
+ */
+#define CLOCK_CORECLOCK     (1000000UL * CONFIG_ESP32S3_DEFAULT_CPU_FREQ_MHZ)
```

Comment:
This would require to include RIOT's `macros.h` which leads to a number of naming conflicts when compiling ESP IDF sources, including the `MHZ` macro. `sdkconfig_*.h` are mainly used for the configuration of ESP-IDF with this only acception. A solution could be to move `CLOCK_CORECLOCK` to `cpu_conf.h`. It's just still there since it was added there with https://github.com/RIOT-OS/RIOT/commit/5ae5c40f26de541c1ac804623520f67442588f83`.

Should I try to move it to `cpu_conf.h` for all ESP32x SoCs with this PR?

Reply:
It would make more sense in `cpu_conf.h` since it's not for configuring the SDK but for RIOT consumption only.

Reply:
Should it be a separate PR which would require an additional rebase or could it be part of this PR?

Reply:
I don't really have a strong opinion there, whatever you prefer.

Reply:
Commit 066239da217be39fff1f79de94f33896b91857ce moves the define to the corresponding `cpu_*.h` files.